docs/imagecustomizer/api/cosi.md

```diff
@@ -14,101 +21,110 @@ The COSI file MUST be an uncompressed tarball. The file extension SHOULD be `.co
 The tarball MUST contain the following files:
 
 - `metadata.json`: A JSON file that contains the metadata of the COSI file.
-- Partition image files in the folder `images`: The actual partition images
-  that will be used to install the OS.
+- Filesystem image files in the folder `images/`: The actual filesystem images
+  that Trident will use to install the OS.
+
+To allow for future extensions, the tarball MAY contain other files, but Trident
+MUST ignore them. The tarball SHOULD NOT contain any extra files that will not
+be used by Trident.
+
+### Layout
 
-If the tarball contains other files, readers MUST ignore them. A writer SHOULD NOT
-add any other files to the tarball.
+The tarball MUST NOT have a common root directory. The metadata file MUST be at
+the root of the tarball. If it were extracted with a standard `tar` invocation,
+the metadata file would be placed in the current directory.
 
-The tarball MUST NOT have a common root directory. The `metadata.json` file and the
-`images` directory MUST be in the root directory of the tarball.
+The metadata file SHOULD, be placed at the beginning of the tarball to allow for
```

Comment:
You should keep this as:

```
The metadata file SHOULD be placed at the beginning of the tarball to allow for
quick access to the metadata without needing to traverse the entire tarball.
```

It might also be good to backport the fixes.